tests/net/lib/coap/src/main.c

```diff
@@ -1855,4 +1855,110 @@ ZTEST(coap, test_age_is_newer)
 		     "Max age should be marked as newer");
 }
 
+struct test_coap_request {
+	uint16_t id;
+	uint8_t token[COAP_TOKEN_MAX_LEN];
+	uint8_t tkl;
+	uint8_t code;
+	enum coap_msgtype type;
+	struct coap_reply *match;
+};
+
+static int reply_cb(const struct coap_packet *response,
+		    struct coap_reply *reply,
+		    const struct sockaddr *from)
+{
+	return 0;
+}
+
+ZTEST(coap, test_response_matching)
+{
+	struct coap_reply matches[] = {
+		{ }, /* Non-initialized (unused) entry. */
+		{ .id = 100, .reply = reply_cb },
+		{ .id = 101, .token = { 1, 2, 3, 4 }, .tkl = 4, .reply = reply_cb },
+	};
+	struct test_coap_request test_responses[] = {
+		/* #0 Piggybacked ACK, no token */
+		{ .id = 100, .type = COAP_TYPE_ACK, .match = &matches[1],
```

Comment:
Maybe too picky, but RFC 7252 named that "empty token" not "no token". In the case of an empty message (code 0.0) the it may be considered as "no token", but for non empty messages the wording is an empty token.

Reply:
Changed to "empty token"